src/mapping/nav2_map_server/src/map_server.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_map_server/map_server.hpp"
+#include "nav2_map_server/map_loader.hpp"
+#include <string>
+
+#define USAGE    "\nUSAGE: map_server <map.yaml> <map_type>\n" \
+  "  map.yaml: map description file\n" \
+  "  map_type: the type of map to load (i.e. occupancy)\n"
+
+
+MappingServerROS::MappingServerROS(const std::string & fname, const std::string & map_type)
```

Comment:
Classes should have their own headers and implementation files, named according to the class. In this case, mapping_server_ros.hpp and mapping_server_ros.cpp, containing the declaration and implementation of MappingServerRos.